### ISSUE TITLE
bash completion doesn't redraw-current-line

### ISSUE THREAD

**Issue by xalexalex** [junegunn/fzf#1134]
- Category
    - [ ] fzf binary
    - [ ] fzf-tmux script
    - [ ] Key bindings
    - [x] Completion
    - [ ] Vim
    - [ ] Neovim
    - [ ] Etc.
- OS
    - [x] Linux
    - [ ] Mac OS X
    - [ ] Windows
    - [ ] Windows Subsystem for Linux
    - [ ] Etc.
- Shell
    - [X] bash
    - [ ] zsh
    - [ ] fish

## Issue

With bash version 4.4.12 and fzf version 0.17.1 (commit 0b33dc6), tab completion doesn't redraw the current line after fzf exits. This means that if I type `mv folder/f**`, and then select `folder/file` in fzf, the line doesn't become `mv folder/file` but `ile ` with a trailing space.

With keybindings, everything works as expected. In zsh v5.4.2, both keybindings and completion work flawlessly.

## Steps to reproduce:

In bash, type `mv **`, then `<Tab>` and in fzf select `/home/user/file`
The line I get after fzf exits is just `file `, with a trailing space. However the full line is present and if I manually `redraw-current-line` (with `\er`, i.e. `<Alt-R>`), it shows up correctly.

Similarly,
`mv fi**`, `<Tab>`, select `/home/user/file`
I get `le `, without the `fi` part that I wrote, which would be `$base` in `__fzf_generic_path_completion()` in completion.bash. As expected, `<Alt-R>` fixes the line.

The same happens with `kill <Tab>` and the other completions.

## Expected behavior:

`mv **`, `<Tab>`, select `/home/user/file`
The line I should get is `mv /home/user/file` (fzf's completion should automatically trigger `redraw-current-line`).

**Comment by xalexalex** [comment]
EDIT: the issue is reproducible on st v0.7 but not on urxvt v9.22 nor termite v12. So it is a st issue, while fzf works fine. I should have checked earlier.

**Comment by komidore64** [comment]
> EDIT: the issue is reproducible on st v0.7 but not on urxvt v9.22 nor termite v12. So it is a st issue, while fzf works fine. I should have checked earlier.

@xalexalex where you able to fix this issue in `st`? I apologize for high-jacking a PR, but this is about the only thing I'm able to find outlining this particular issue that I'm currently facing.

**Comment by xalexalex** [comment]
Nope, the issue is still there with st v0.8.4 (actually on the current HEAD which is 4536f46cff) and fzf v0.27.2 (8255aa2).

It's strange because it should be a fzf or readline issue, while the terminal shouldn't be involved in this kind of stuff—but apparently it is. You might want to try to file a bug report in the appropriate suckless mailing list.

**Comment by komidore64** [comment]
Apologies for another necro, but I just wanted to provide [some more information](https://www.reddit.com/r/suckless/comments/f9saa1/bind_e0n_in_st/) about why this doesn't work in `st` but works in other terms like `xterm` for anyone who finds this thread in the future.

tl;dr
> `st` does not support those escape sequences.
